### PR TITLE
Extract query parsing code into a separate class and introduce constants

### PR DESCRIPTION
**Pull request by skofman1**
@maartenba , @xavierdecoster

### CODE REVIEW

File: src/NuGet.Indexing/NuGetQueryParser.cs [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+using System.Linq;
+using System.Text;
+
+namespace NuGet.Indexing
+{
+    public class NuGetQueryParser
+    {
+        private readonly Dictionary<QueryField, string[]> _queryFieldNames = new Dictionary<QueryField, string[]>
+        {
+            {QueryField.Id,  new [] { "Id" }},
```
Comment (maartenba) [inline src/NuGet.Indexing/NuGetQueryParser.cs]: Is the uppercase "I" intended here?

File: src/NuGet.Indexing/NuGetQueryParser.cs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            {QueryField.PackageId,  new [] { "packageid" }},
```
Comment (maartenba) [inline src/NuGet.Indexing/NuGetQueryParser.cs]: Also add some spaces :-)  (`{ QueryField...`)

File: src/NuGet.Indexing/NuGetQueryParser.cs [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                        yield return new Clause { Field = "*", Text = token.Value };
+                    }
+                    field = null;
+                }
+            }
+
+            yield break;
+        }
+
+        private static IEnumerable<Token> Tokenize(string s)
```
Comment (maartenba) [inline src/NuGet.Indexing/NuGetQueryParser.cs]: Var all the things

**Comment by maartenba** [comment]
I like it! :shipit:

Can you rebase this on the `dev` branch? The `searchcloud` branch should be removed :-)

File: src/NuGet.Indexing/NuGetQuery.cs [hunk 4]
```diff
@@ -21,17 +21,8 @@ public static Query MakeQuery(string q)
...
-            {
-                HashSet<string> text;
-                if (!grouping.TryGetValue(clause.Field, out text))
-                {
-                    text = new HashSet<string>();
-                    grouping.Add(clause.Field, text);
-                }
-                text.Add(clause.Text);
-            }
+            var queryParser = new NuGetQueryParser();
```
Comment (xavierdecoster) [inline src/NuGet.Indexing/NuGetQuery.cs]: Coolness!

**Comment by xavierdecoster** [comment]
:shipit:

File: src/NuGet.Indexing/NuGetQueryParser.cs (hunk 1 above)
Comment (skofman1) [inline src/NuGet.Indexing/NuGetQueryParser.cs]: no.. changed to lower case

---
In reply to: [64862493](https://github.com/NuGet/NuGet.Services.Metadata/pull/73#discussion_r64862493) [](ancestors = 64862493)

**Comment by skofman1** [comment]
merged into dev :)

**Pull request closed without merging**